### PR TITLE
Update tests for quarkus 3.15

### PR DESCRIPTION
**Pull request by mocenas**
### Summary

Reopening #1980 Against 3.15 branch.
Test updates via quarkus CLI from version 3.8 to 3.15.

Please select the relevant options.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Dependency update
- [ ] Refactoring
- [ ] Backport
- [X] New scenario (non-breaking change which adds functionality)
- [ ] This change requires a documentation update
- [ ] This change requires execution against OCP (use `run tests` phrase in comment)

### Checklist:
- [X] Methods and classes used in PR scenarios are meaningful
- [X] Commits are well encapsulated and follow [the best practices](https://cbea.ms/git-commit/)

### CODE REVIEW

File: quarkus-cli/src/test/java/io/quarkus/ts/quarkus/cli/update/AbstractQuarkusCliUpdateIT.java [hunk 1]
```diff
@@ -21,37 +34,28 @@
...
     protected final DefaultArtifactVersion oldVersionStream;
     protected final DefaultArtifactVersion newVersionStream;
-    protected final String newVersionFromProperties;
     protected final IQuarkusCLIAppManager quarkusCLIAppManager;
 
     public AbstractQuarkusCliUpdateIT(DefaultArtifactVersion oldVersionStream, DefaultArtifactVersion newVersionStream) {
         this.oldVersionStream = oldVersionStream;
         this.newVersionStream = newVersionStream;
 
-        // takes quarkus.platform.version from maven parameters. If present, it will update to this exact BOM version
```
[the remark this answers is not in the document]
Comment (mocenas) [reply quarkus-cli/src/test/java/io/quarkus/ts/quarkus/cli/update/AbstractQuarkusCliUpdateIT.java:43]: We don't need to directly access this here anymore. Getting maven's quarkus version is wrapped either in `QuarkusProperties.isRHBQ` or in `QuarkusCLIUtils.createAppManager`. Which will get maven's quarkus where they need it.

**Review by fedinskiy — APPROVED** [review]
@mocenas the run in CI was disabled for some reason:
`2024-09-18T08:16:49.2096876Z [WARNING] Tests run: 15, Failures: 0, Errors: 0, Skipped: 15, Time elapsed: 0.001 s -- in io.quarkus.ts.quarkus.cli.update.QuarkusCli38to315UpdateIT`

Could you take a look?

**Comment by mocenas** [comment]
> ?


> @mocenas the run in CI was disabled for some reason: `2024-09-18T08:16:49.2096876Z [WARNING] Tests run: 15, Failures: 0, Errors: 0, Skipped: 15, Time elapsed: 0.001 s -- in io.quarkus.ts.quarkus.cli.update.QuarkusCli38to315UpdateIT`
> 
> Could you take a look?

These tests have condition to only run on quarkus 3.15. But current pom version is only 3.14.999 so tests will not run.
Since the tests are going only to 3.15 branch, I can remove this condition.

**Comment by fedinskiy** [comment]
> Since the tests are going only to 3.15 branch, I can remove this condition.

yes, please, there is not much difference between 3.14 and 3.15, but we need to have working autotests

**Comment by mocenas** [comment]
> > Since the tests are going only to 3.15 branch, I can remove this condition.
> 
> yes, please, there is not much difference between 3.14 and 3.15, but we need to have working autotests

Already done, just now waiting for CI.

**Pull request merged**